### ISSUE TITLE
Support CompletionItem.textEdit

### ISSUE THREAD

**Issue by MaskRay** [ul/kak-lsp#248]
https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/

```typescript
	/**
	 * An edit which is applied to a document when selecting this completion. When an edit is provided the value of
	 * `insertText` is ignored.
	 *
	 * *Note:* The range of the edit must be a single line range and it must contain the position at which completion
	 * has been requested.
	 */
	textEdit?: TextEdit
```

See https://discuss.kakoune.com/t/kak-lsp-ccls/750 for more information.

**Comment by ul** [comment]
#40 — we are waiting on some changes in Kakoune itself to support it properly.

**Comment by MaskRay** [comment]
Thanks. I'll keep an eye on  #40 and mawww/kakoune#2898.

Duplicate of #40.
